apps/cyberstorm-nextjs/utils/auth.ts

```diff
@@ -0,0 +1,132 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import hkdf from "@panva/hkdf";
+import { EncryptJWT, jwtDecrypt } from "jose";
+import { v4 as uuid } from "uuid";
+import { isRecord } from "./typeChecks";
+
+interface UserStatusType {
```

Comment:
As mentioned in some earlier comment or discord discussion, when taking this approach with the type guard, this could be named `PartialUserStatus` oslt, to make it obvious for other developers that not all properties are type checked and they shouldn't assume so or their foot might get new ventilation holes.

Obviously also remove the props that are not checked from the interface.

Reply:
Ah, true true, seems like I missed or forgot that part of the comment/discussions earlier. I'll change these up 💯